conda_smithy/templates/circle.yml.tmpl

```diff
@@ -24,14 +24,31 @@ dependencies:
   override:
     - docker pull condaforge/linux-anvil
 
+{%- if matrix %}
+{%- if 'ccache-toolchain' in ' '.join(package['meta']['requirements']['build']) %}
+    # Run, test and (if we have a BINSTAR_TOKEN) upload the distributions.
+    - ./ci_support/run_docker_build.sh || touch ./build_failed
+  cache_directories:
+    - "./build_artefacts/.ccache"
+{% else %}
+{% endif -%}
+{% endif -%}
+
 {% endif -%}
 {% endblock -%}
+
 test:
   override:
 {%- block env_test %}
 {%- if matrix %}
+
+{%- if 'ccache-toolchain' in ' '.join(package['meta']['requirements']['build']) %}
+    - if test -f ./build_failed; then exit 1; fi
```

Comment:
It would be better to echo a message before exiting:

```if test -f ./build_failed; echo "Build step failed. Please look at the exact error message above"; then exit 1; fi```